docs/documentation/automation/admin_cotlang.md

```diff
@@ -7,18 +7,236 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+  ...COTAnswer,
+  messages: COTMessage
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
 
-#### Coming soon!
+**Note**: _The source is a bot that is triggered with a slash command and is associated with particular channels:_ `/command`
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### Channel Survey Trigger {#channel-survey-trigger}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+    ...COTAnswer,
+    messages: COTMessage
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**: _Messages have to be active to use them. The source is a bot that is triggered with a survey and is associated with particular channels._
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### SM Survey Trigger {#sm-survey-trigger}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+    ...COTTask,
+    sentAnswer: COTAnswer
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**: 
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### SM Change State {#sm-change-state}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+    ...COTTASK
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**: 
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### SM - New Subtask {#new-subtask}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+    task: COTTask,
+    parent: COTTask,
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**: 
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### SM - New Task {#new-task}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+    task: COTTask,
+    parent: COTTask
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**: 
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### Service-Level Agreement (SLA) {#sla}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+    taskId: ObjectId[COTTask],
+    taskGroupId: ObjectId[COTTaskGroup]
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**: 
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### Scheduler {#scheduler}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+_custom body_
+
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**:
+
+</div>
+</div>
+
+<div className="row table-row-1">
+
+### SM - Required Survey {#required-survey}
+
+<div className="col col--11 col--offset-1 code-table-1">
+
+```typescript
+{
+    answer: COTAnswer,
+    meta: {
+        parentTask: ObjectId[COTTask],
+        taskGroup: ObjectId[COTTaskGroup]
+    }
+}
+```
+</div>
+<div className="col col--11 col--offset-1 code-table-2">
+
+**Note**: _notes_
```

Comment:
note: notes

Reply:
**Requied Surveys** are events that are triggered **before** a new task, sub-task is created or a state-change will be performed. They **require** a survey to be enabled.

Reply:
Should we stick with the term "required survey"? What did you think of the proposal "Task Procedure Form"? Or are there any other suggestions?

Reply:
1. "Procedure" in computing this has a special meaning (It used as synonymous function or method) I feel uneasy when reading task-procedure-form. 
2. Also, I think, "procedure" is too generic - anything in cotalker could be referenced as a procedure. 
3. I don't like Required Surveys - the required surveys is a side effect of the main purpose - we want start (initial-forms) or move (change-states) tasks with certain rules and triggers :  e.g.,   To create a task, you require to state when it will be needed or to close the task you require to say which was the cost of the project.
4. I also believe we need to split them up in two categories : pre-creation-trigger and pre-state-change-trigger

Reply:
Looking at the other triggers we have some that run after the task is created and changed state

Reply:
Perhaps we can call them something in the tone of 

Pre-creation-trigger
Post-creation-trigger
Pre-change-state-trigger
Post-change-state-trigger

So the names are very functional

Reply:
@rafaelwalter does this make sense with the implementation

Reply:
# Ideas for a new *required survey* name:
_My personal favorites are in **bold font**._


| Main Title | Pre-creation Trigger | Post-creation Trigger | Pre-change-state-trigger | Post-change-state-trigger |
| ---- | ---- | ---- | ---- | ----- | 
| Task Forms | Initialize-task form | Initialized-task form | Change-state task form | Changed-state task form |
| Task Trigger Forms | Pre-Initialization trigger form | Post-initialization trigger form | Pre-change-state trigger form | Post-change-state trigger form|
| Task Forms | Preliminary task initialization setup form | Subsequent initialized task form | Preliminary Change-task-state form  | Subsequent Change-task-state form |
| **Task Initialization Forms** | **Preliminary Task Initialization Form** | **Post-Task-Initialization Form** | | |
| Task Modification Forms | | | Preliminary Task Modification Form | Post-Task-Modification Form | 
| **Task-State-Change Forms** |  |  | **Preliminary Task-State-Change Form** |  **Post-Task-State-Change Form** | 
| Task-State-Modification Form |  |  | Preliminary Task-State-Modification Form | Post-Task-Modification-Form |

Reply:
Oh, pre and post are standard programming names for events e.g., a quick search https://www.ibm.com/docs/en/powerha-aix/7.2?topic=events-configuring-pre-event-post-event-commands

Reply:
Great table btw

Reply:
This table makes a lot of sense about what we should consider.

@Mauriceac perhaps we should look at the new automation tools we are working at. It would be great to unify this.